javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -467,52 +467,16 @@ public Short getNext() {
      * @param from        the first character
      * @param toExclusive the successor of the last character if step &gt; 0, the predecessor of the last character if step &lt; 0
      * @param step        the step
-     * @return a range of characters as specified or the empty range if {@code (from == toExclusive) || (step * (from - toExclusive) > 0)}.
+     * @return a range of characters as specified or the empty range if {@code signum(step) == signum(from - toExclusive)}.
      * @throws IllegalArgumentException if {@code step} is zero
      */
     static Iterator<Character> rangeBy(char from, char toExclusive, int step) {
         return Iterator.rangeBy((int) from, (int) toExclusive, step).map(i -> (char) i.shortValue());
     }
 
     static Iterator<Double> rangeBy(double from, double toExclusive, double step) {
-        if (Double.isNaN(from)) {
-            throw new IllegalArgumentException("from is NaN");
-        } else if (Double.isNaN(toExclusive)) {
-            throw new IllegalArgumentException("toExclusive is NaN");
-        } else if (Double.isNaN(step)) {
-            throw new IllegalArgumentException("step is NaN");
-        } else if (step == 0) {
-            throw new IllegalArgumentException("step cannot be 0");
-        } else if (step * (from - toExclusive) >= 0) {
-            return Iterator.empty();
-        } else {
-            return new AbstractIterator<Double>() {
-
-                double prev = Double.NaN;
-                double curr = from;
-                boolean hasNext = true;
-
-                @Override
-                public boolean hasNext() {
-                    return hasNext;
-                }
-
-                @Override
-                public Double getNext() {
-                    final double next = curr;
-                    if ((step > 0 && curr + step >= toExclusive) || (step < 0 && curr + step <= toExclusive)) {
-                        hasNext = false;
-                    } else {
-                        prev = curr;
-                        curr += step;
-                        if (curr == prev) {
-                            hasNext = false;
-                        }
-                    }
-                    return next;
-                }
-            };
-        }
+        final double toInclusive = (step > 0) ? Math.nextDown(toExclusive) : Math.nextUp(toExclusive);
```

Comment:
make it inclusive by incrementing it with a `ULP`

Note: this will make exclusive infinities not possible (had to modify a test for that), but I think an exclusive equality to infinity is strange anyway